docs/terms/plugin.md

```diff
@@ -0,0 +1,63 @@
+
+### Component specific functionality
+
+Because Plugins often extend the functionality of existing Greate Expectations components, it is impossible to classify all of their potential features in a few generic statements.  In general, best practices are to include thorough documentation if you are developing or contributing code for use as a Plugin.  If you are using code that was created by someone else, you will have to reference their documentation (and possibly their code itself) in order to determine the features of that specific Plugin.
+
+## API basics
+
+The API of any given Plugin is determined by the individual or team that created it.  That said, if the Plugin is extending an existing Great Expectations component, then best practices are for the Plugin's API to mirror that of the object it extends as closely as possible.
+
+### Importing
+
+Any Plugin dropped into the `plugins` folder can be imported with a standard Python import statement.  In some cases, this will be all you need to do in order to make use of the Plugin's functionality.  For example, a <TechnicalTag relative="../" tag="custom_expectation" text="Custom Expectation" /> Plugin could be imported and used the same as any other Expectation in the [interactive process for creating Expectations](../guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md).
+
+### Configuration
+
+If a Plugin can't be directly used from an import, it can typically be used by editing the relevant configuration file to reference it.  This typically involves setting the `module_name` for an object to the module name of the Plugin (as you would type it in an import statement) and the `class_name` for that same object to the class name that is implemented in the Plugin file.
+
+For example, say you created a pair of Plugins called `my_custom_dataset.py` and `my_custom_datasource.py` that implement the classes `MyCustomDataset` and  `MyCustomDatasource` respectively.  To use those Plugins to connect to data, you would edit the configuration for the corresponding <TechnicalTag relative="../" tag="datasource" text="Datasource" /> in your `great_expectations.yml` file to contain an entry like the following:
+
+```yaml
+datasources:
+  db_name:
+    credentials: ${db_name}
+    data_asset_type:
+      class_name: MyCustomDataset
+      module_name: my_custom_dataset
+    class_name: MyCustomDatasource
+    module_name: MyCustomDatasource
```

Comment:
Love including an example here! Maybe we should use a v3 api example e.g.:
```suggestion
```yaml
datasources:
  my_datasource:
    execution_engine:
      class_name: SqlAlchemyExecutionEngine
      module_name: great_expectations.execution_engine
      connection_string: ${my_connection_string}
    data_connectors:
      my_custom_data_connector:
        class_name: MyDataConnector
        module_name: my_data_connector
```

Reply:
Thanks -- yes, I based the original one off of the stack overflow question your directed me to.

@talagluck recommended we use a custom execution engine and custom data connector. Your updated example seems to only change the data connector to a custom class, correct?  I'll update the text leading to this example to indicate that only a custom data connector is being created, as well.

Reply:
Correct! I think adding a custom data connector is a more likely scenario, but either technically works. Thanks for updating!